src/utils/identity/ledger/index.ts

```diff
@@ -0,0 +1,159 @@
+import { SignIdentity, PublicKey, HttpAgentRequest, ReadRequest, CallRequest, Cbor } from '@dfinity/agent';
+import { BinaryBlob, blobFromBuffer} from '@dfinity/candid';
+import { Principal } from '@dfinity/principal';
+import DfinityApp, { ResponseSign } from '@zondax/ledger-icp';
+
+import { JsonnableKeyPair } from '../../../interfaces/identity';
+import { base64ToBytes, bytesToBase64 } from '../../encoding';
+import { GenericSignIdentity } from '../genericSignIdentity';
+import Secp256k1PublicKey from '../secpk256k1/publicKey';
+import { Buffer as BlobBuffer } from '../../../../node_modules/buffer';
+
+/**
+ * Convert the HttpAgentRequest body into cbor which can be signed by the Ledger Hardware Wallet.
+ * @param request - body of the HttpAgentRequest
+ */
+ export function _prepareCborForLedger(request: ReadRequest | CallRequest): BinaryBlob {
+  return Cbor.encode({ content: request });
+}
+class LedgerIdentity extends SignIdentity implements GenericSignIdentity {
+
+  protected derivePath: string;
+  protected publicKey: PublicKey;
+  protected TransportClass: any;
+
+  public static async create(derivePath = `m/44'/223'/0'/0/0`, TransportClass: any): Promise<LedgerIdentity> {
+    const transport = await TransportClass.create();
+    const app = new DfinityApp(transport);
+
+    const resp = await app.getAddressAndPubKey(derivePath);
+    // This type doesn't have the right fields in it, so we have to manually type it.
+    const principal = (resp as unknown as { principalText: string }).principalText;
+    const publicKey = Secp256k1PublicKey.fromRaw(blobFromBuffer(new BlobBuffer(resp.publicKey)));
+
+    if (principal !== Principal.selfAuthenticating(new Uint8Array(publicKey.toDer())).toText()) {
+      throw new Error('Principal returned by device does not match public key.');
+    }
+
+    transport.close()
+
+    return new this(derivePath, publicKey, TransportClass);
+  }
+
+  public static async fromJSON(json: string, TransportClass: any): Promise<LedgerIdentity> {
+    const parsed = JSON.parse(json);
+    if (Array.isArray(parsed)) {
+      if (typeof parsed[0] === 'string' && typeof parsed[1] === 'string') {
+        const derPublicKey = base64ToBytes(parsed[1]);
+        const publicKey = Secp256k1PublicKey.fromDer(blobFromBuffer(new BlobBuffer(derPublicKey)));
+        return new LedgerIdentity(parsed[0], publicKey, TransportClass);
+      }
+    }
+    throw new Error(
+      'Deserialization error: JSON must have at least 2 items.'
+    );
+  }
+  
+  constructor(derivePath = `m/44'/223'/0'/0/0`, publicKey: PublicKey, TransportClass: any) {
```

Comment:
we have the path in a constant in the accounts utils

Reply:
Yep